src/main/resources/templates/project/project-add.html

```diff
@@ -13,23 +13,41 @@ <h2>Add project details</h2>
 
 <div th:if="${errorMessage}" th:utext="${errorMessage}" class="alert alert-danger" role="alert"></div>
 
-<form class="row g-3" th:action="@{/projects/add}" th:object="${project}" method="post">
+<form class="row g-3 needs-validation" novalidate action="#" th:action="@{/projects/add}" th:object="${project}" method="post">
+
+    <div>
+        <div class="col-md-4">
+            <label for="name" class="form-label">Name<a style="color:red">*</a></label>
```

Comment:
add * sign like here http://blog.adeel.io/2015/10/18/adding-an-asterisk-to-required-fields-in-bootstrap/